include/mppic/utils.hpp

```diff
@@ -15,11 +15,22 @@
 #include "geometry_msgs/msg/twist_stamped.hpp"
 #include "nav_msgs/msg/path.hpp"
 #include "nav2_util/node_utils.hpp"
-#include "mppic/optimization/tensor_wrappers/control_sequence.hpp"
+#include "mppic/tensor_wrappers/control_sequence.hpp"
 
 namespace mppi::utils
 {
 
+struct VelocityConstraints
+{
+  VelocityConstraints(const double x, const double y, const double w)
+  : vx(x), vy(y), vw(w)
+  {}
+
+  VelocityConstraints() {}
+
+  double vx, vy, vw;
+};
+
```

Comment:
```suggestion
struct VelocityConstraints
{
  double vx; 
  double vy;
  double vw;
};

```

Reply:
Hmm... May be control constraints ? i dont like this VelocityConstraints.velocityx

Reply:
I can do control constraints, sure. I would like to keep these constructors though, they make some of the code much cleaner for setting values in 1 line.

Reply:
constuctors will automatically generated )

Reply:
```
/home/steve/Documents/nav2_ws/src/mppic/src/controller.cpp: In member function ‘virtual void mppi::Controller::setSpeedLimit(const double&, const bool&)’:
/home/steve/Documents/nav2_ws/src/mppic/src/controller.cpp:89:86: error: no matching function for call to ‘mppi::utils::ControlConstraints::ControlConstraints(double&, double&, double&)’
   89 |     constraints = utils::ControlConstraints(base_x_vel_, base_y_vel_, base_theta_vel_);
      |                                                                                      ^
In file included from /home/steve/Documents/nav2_ws/src/mppic/include/mppic/critic_manager.hpp:15,
                 from /home/steve/Documents/nav2_ws/src/mppic/include/mppic/optimizer.hpp:15,
                 from /home/steve/Documents/nav2_ws/src/mppic/include/mppic/controller.hpp:5,
                 from /home/steve/Documents/nav2_ws/src/mppic/src/controller.cpp:3:
/home/steve/Documents/nav2_ws/src/mppic/include/mppic/utils.hpp:23:8: note: candidate: ‘mppi::utils::ControlConstraints::ControlConstraints()’
   23 | struct ControlConstraints
      |        ^~~~~~~~~~~~~~~~~~
/home/steve/Documents/nav2_ws/src/mppic/include/mppic/utils.hpp:23:8: note:   candidate expects 0 arguments, 3 provided
/home/steve/Documents/nav2_ws/src/mppic/include/mppic/utils.hpp:23:8: note: candidate: ‘constexpr mppi::utils::ControlConstraints::ControlConstraints(const mppi::utils::ControlConstraints&)’
/home/steve/Documents/nav2_ws/src/mppic/include/mppic/utils.hpp:23:8: note:   candidate expects 1 argument, 3 provided
/home/steve/Documents/nav2_ws/src/mppic/include/mppic/utils.hpp:23:8: note: candidate: ‘constexpr mppi::utils::ControlConstraints::ControlConstraints(mppi::utils::ControlConstraints&&)’
/home/steve/Documents/nav2_ws/src/mppic/include/mppic/utils.hpp:23:8: note:   candidate expects 1 argument, 3 provided
/home/steve/Documents/nav2_ws/src/mppic/src/optimizer.cpp: In member function ‘mppi::utils::ControlConstraints mppi::Optimizer::getControlConstraints()’:
/home/steve/Documents/nav2_ws/src/mppic/src/optimizer.cpp:296:61: error: no matching function for call to ‘mppi::utils::ControlConstraints::ControlConstraints(double&, double&, double&)’
  296 |   return utils::ControlConstraints(vx_max_, vy_max_, wz_max_);
      |                                                             ^
In file included from /home/steve/Documents/nav2_ws/src/mppic/include/mppic/critic_manager.hpp:15,
                 from /home/steve/Documents/nav2_ws/src/mppic/include/mppic/optimizer.hpp:15,
                 from /home/steve/Documents/nav2_ws/src/mppic/src/optimizer.cpp:1:
/home/steve/Documents/nav2_ws/src/mppic/include/mppic/utils.hpp:23:8: note: candidate: ‘mppi::utils::ControlConstraints::ControlConstraints()’
   23 | struct ControlConstraints
      |        ^~~~~~~~~~~~~~~~~~
/home/steve/Documents/nav2_ws/src/mppic/include/mppic/utils.hpp:23:8: note:   candidate expects 0 arguments, 3 provided
/home/steve/Documents/nav2_ws/src/mppic/include/mppic/utils.hpp:23:8: note: candidate: ‘constexpr mppi::utils::ControlConstraints::ControlConstraints(const mppi::utils::ControlConstraints&)’
/home/steve/Documents/nav2_ws/src/mppic/include/mppic/utils.hpp:23:8: note:   candidate expects 1 argument, 3 provided
/home/steve/Documents/nav2_ws/src/mppic/include/mppic/utils.hpp:23:8: note: candidate: ‘constexpr mppi::utils::ControlConstraints::ControlConstraints(mppi::utils::ControlConstraints&&)’
/home/steve/Documents/nav2_ws/src/mppic/include/mppic/utils.hpp:23:8: note:   candidate expects 1 argument, 3 provided

```

Maybe in an older C++ version or with some compiler flags shut off? This gives me errors and asks for a constructor

Reply:
Use braced initialization {}